earthfile2llb/converter.go

```diff
@@ -1828,7 +1828,9 @@ func (c *Converter) prepBuildTarget(ctx context.Context, fullTargetName string,
 	}
 	// Don't allow transitive overriding variables to cross project boundaries (unless --pass-args is used).
 	propagateBuildArgs := !relTarget.IsExternal()
-	if passArgs || propagateBuildArgs {
+	if passArgs {
+		overriding = variables.CombineScopes(overriding, c.varCollection.Overriding(), c.varCollection.Args(), c.varCollection.Globals())
+	} else if propagateBuildArgs || passArgs {
```

Comment:
This condition seems a bit confusing as it won't be executed if `passArgs` is true given L1831.

Reply:
good catch! that slipped back in when I was rebasing.